### PR TITLE
test: event sync

### PR DESCRIPTION
**Pull request by CHr15F0x**
Part of #1944

### CODE REVIEW

File: crates/p2p/src/client/peer_agnostic.rs [hunk 1]
```diff
@@ -46,6 +47,26 @@ impl<T> PeerData<T> {
...
+}
+
+impl<T, U: Dummy<T>> Dummy<T> for PeerData<U> {
+    fn dummy_with_rng<R: rand::prelude::Rng + ?Sized>(config: &T, rng: &mut R) -> Self {
+        let digest = rng.gen::<[u8; 32]>();
+        let multihash = libp2p::multihash::Multihash::wrap(0x0, &digest)
+            .expect("The digest size is never too large");
+
+        PeerData {
+            peer: PeerId::from_multihash(multihash).expect("Valid multihash"),
```
Comment (Mirko-von-Leipzig) [inline crates/p2p/src/client/peer_agnostic.rs:66]: Btw I discovered today there is a `PeerId::random()`

Comment (Mirko-von-Leipzig) [inline crates/p2p/src/client/peer_agnostic.rs:66]: Why not just use `PeerId::random()` here as well?

**Review by Mirko-von-Leipzig — APPROVED** [review]
I like it; only concern is how we generate this data. But I don't really have an ergonomic suggestion either so :)

**Comment by CHr15F0x** [comment]
> only concern is how we generate this data.

I'm not happy with it either but so far haven't had good ideas either.

**Pull request merged**